src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionsUtilityToolServlet.java

```diff
@@ -724,7 +724,7 @@ private String extractNextLink(HttpResponse canvasResponse) {
 		if(links.containsKey(searchPhrase)){
 			M_log.debug("LINKS CONTAINS NEXT: " + links.get(searchPhrase));
 			linkValueNext = "/canvasCourseManager/manager" + links.get(searchPhrase).substring(links.get(searchPhrase).indexOf("/api"), links.get(searchPhrase).length());
-			if(linkValueNext.contains("as_user_id")){
+			if(linkValueNext.contains("as_user_id") && session.getAttribute(SectionUtilityToolFilter.LAUNCH_TYPE) == "lti"){
 				linkValueNext = linkValueNext.replaceAll("as_user_id=sis_login_id.*?&", "user=self&");
```

Comment:
The `"lti"` string can be made as constant since it is being referenced more than one place in the code.